cfme/common/__init__.py

```diff
@@ -126,3 +119,50 @@ def validate_properties(self):
             assert(expected_value == shown_value,
                    'Property {} has wrong value, expected {} but was {}'
                    .format(property_tuple[1], expected_value, shown_value))
+
+
+class SummaryPage(object):
+    """
+    Middleware provider and other pages in middleware must extend this class
+    to get table details of `Properties`, `Relationships` and `Smart Management`
+    """
+
+    def properties_ui(self):
+        """
+        `Properties` table
+        Returns: dict value of table details
+        """
+        return self.summary_data_ui("Properties")
+
+    def relationships_ui(self):
+        """
+        `Relationships` table
+        Returns: dict value of table details
+        """
+        return self.summary_data_ui("Relationships")
+
+    def smart_management_ui(self):
+        """
+        `Smart Management` table
+        Returns: dict value of table details
+        """
+        return self.summary_data_ui("Smart Management")
+
+    def summary_data_ui(self, header):
```

Comment:
`properties_ui`, `relationships_ui` and `smart_management_ui` methods should be just `functools.partial(self.get_detail, <table_header>)` where <table_header> is the header of the infoblock section where you want to fetch data from. I dont believe there is a need for this `summary_data_ui` method; tell me if Im missing something ;)